cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -299,28 +301,56 @@ func findOVAFiles(directory string) ([]string, []string, error) {
 	}
 
 	var ovaFiles, ovfFiles []string
+
+	// Check for .ova files in the main directory (zero level)
+	for _, child := range childs {
+		path := filepath.Join(directory, child.Name())
+		if hasSuffixIgnoreCase(child.Name(), OvaExt) {
+			ovaFiles = append(ovaFiles, path)
+		}
+	}
+
+	// Check for .ova and .ovf files in child directories (first level)
 	for _, child := range childs {
 		if !child.IsDir() {
 			continue
 		}
-		newDir := directory + "/" + child.Name()
+		newDir := filepath.Join(directory, child.Name())
 		files, err := os.ReadDir(newDir)
 		if err != nil {
 			return nil, nil, err
 		}
 		for _, file := range files {
-			path := filepath.Join(directory, child.Name(), file.Name())
+			path := filepath.Join(newDir, file.Name())
 			switch {
-			case strings.HasSuffix(strings.ToLower(file.Name()), ".ova"):
+			case hasSuffixIgnoreCase(file.Name(), OvaExt):
 				ovaFiles = append(ovaFiles, path)
-			case strings.HasSuffix(strings.ToLower(file.Name()), ".ovf"):
+			case hasSuffixIgnoreCase(file.Name(), OvfExt):
 				ovfFiles = append(ovfFiles, path)
+			case file.IsDir():
```

Comment:
what if there's a folder named `arik.ova`?
I'd expect this to be the first check

Reply:
moved to the beginning.